lib/logger/lib/logger.ex

```diff
@@ -799,7 +787,9 @@ defmodule Logger do
          end}
       end
 
-    if compile_time_purge_matching?(compile_metadata) do
+    compile_level = if is_atom(level), do: level, else: :error
```

Comment:
When is `level` not an atom?

Reply:
It seems you can pass anything to the `Logger.log/3` macro. [Before it was captured by the guards in `Logger.bare_log/3`](https://github.com/elixir-lang/elixir/commit/f17c1a68819cdbdb9cdf048b3f4c10b8fe8269f5#diff-e595abf4d143c9f920f1d59b61df7c35R769), but we are using the value of `level` passed in `Logger.log/3` here.